docs/asyncapi.yaml

```diff
@@ -37,171 +39,120 @@ components:
         line5:
-          type: string
-          description: The fifth line of the address
-          example: The County
+          $ref: '#/components/schemas/County'
         postcode:
-          type: string
-          description: The postcode
-          example: AB1 2CD
-    Scheme:
-      type: object
-      required:
-      - name
-      - shortName
-      - year
-      - frequency
-      - agreementNumber
-      properties:
-        name:
-          type: string
-          description: The name of the scheme
-          example: Sustainable Farming Incentive
-        shortName:
-          type: string
-          description: The short name of the scheme
-          example: SFI
-        year:
-          type: string
-          description: The marketing year
-          example: 2022
-        frequency:
-          type: string
-          description: The frequency of payments
-          example: Quarterly
-        agreementNumber:
-          type: string
-          description: The agreement number
-          example: SFI1234567
-    Payment:
-      type: object
-      required:
-      - invoiceNumber
-      - reference
-      - dueDate
-      - settled
-      - calculated
-      - value
-      - period
-      properties:
-        invoiceNumber:
-          type: string
-          description: Invoice number of payment request
-          example: S0000001C000001V001
-        reference:
-          type: string
-          description: Settlement reference of payment request
-          example: PY1234567
-        dueDate:
-          type: string
-          description: The due date of the payment request
-          example: 1 July 2022
-        settled:
-          type: string
-          description: The settled date of the payment request
-          example: 1 July 2022
-        calculated:
-          type: string
-          description: The calculated date of the payment request
-          example: 1 July 2022
-        value:
-          type: string
-          description: The value of the payment
-          example: 1000.00
-        period:
-          type: string
-          description: The period of the payment
-          example: July 2022 to September 2022
+          $ref: '#/components/schemas/Postcode'
+    Street:
+      type: string
+      description: The first line of the address
+      example: 1 Some Street
+    Area:
+      type: string
+      description: The second line of the address
+      example: Some Area
+    Town:
+      type: string
+      description: The third line of the address
+      example: Some Town
+    Region:
+      type: string
+      description: The fourth line of the address
+      example: Some Region
+    County:
+      type: string
+      description: The fifth line of the address
+      example: Some County
+    Postcode:
+      type: string
+      description: The postcode
+      example: FA1 2PY
     Funding:
+      additionalProperties: false
       type: object
       required:
```

Comment:
Level and Rate are marked as not required, just dbl checking this is correct?

I believe level isn't due to Moorlands stuff, maybe rate comes into tht too?

Reply:
Again, not sure. That's just how it was labelled before I started working on the ticket

Reply:
Yeah, unfort'ly this `async` doc hasn't been created 100% accurately.

Acc'ly looking at the schemas in this application code it would suggest that v few fields are required -- hmmmm

Reply:
I'm going to put both down as required just in case

Reply:
I found out that level, area and rate are actually all optional, so I'm going to change those now